src/server/services/process-upload.js

```diff
@@ -1,124 +1,86 @@
 /* eslint camelcase: 0 */
 
-let log = () => {}
-if (process.env.VERBOSE) {
-  log = console.dir
+const path = require('path')
+const { mkdir, writeFile } = require('fs/promises')
+
+const xlsx = require('xlsx')
+
+const knex = require('../db/connection')
+const reportingPeriods = require('../db/reporting-periods')
+const { createUpload } = require('../db/uploads')
+const { createDocuments } = require('../db/documents')
+
+const { UPLOAD_DIR } = require('../environment')
+
+const { ValidationError } = require('./validate-upload')
+
+const normalizeSheetName = (sheetName) => sheetName.trim().toLowerCase()
+
+const uploadFSName = (upload) => {
+  const filename = `upload-id-${upload.id}${path.extname(upload.filename)}`
+  return path.join(UPLOAD_DIR, filename)
 }
 
-const {
-  agencyByCode,
-  user,
-  createUpload,
-  createDocuments,
-  deleteDocuments,
-  getProject,
-  transact
-} = require('../db')
-
-const FileInterface = require('../lib/server-disk-interface')
-const fileInterface = new FileInterface()
-const { validateUpload } = require('./validate-upload')
-
-const processUpload = async ({
-  filename,
-  user_id,
-  agency_id,
-  data,
-  reporting_period_id = null
-}) => {
-  log(`processUpload(): filename is ${filename}`)
-
-  const {
-    valog,
-    documents,
-    spreadsheet,
-    fileParts,
-    reportingPeriod
-  } = await validateUpload({
-    filename,
-    user_id,
-    agency_id,
-    data,
-    reporting_period_id
-  })
+async function extractDocuments (buffer) {
+  const workbook = await xlsx.read(buffer, { type: 'buffer' })
 
-  if (!valog.success()) {
-    log('valog.success() is false')
-    return { valog, upload: {} }
+  // get begin generating (partial) document rows
+  const documents = []
+  for (const sheetName of workbook.SheetNames) {
+    documents.push({
+      type: normalizeSheetName(sheetName),
+      content: xlsx.utils.sheet_to_json(workbook.Sheets[sheetName], { header: 1 })
```

Comment:
@ajhyndman -- i'm not sure if this is correct/preferred, but here i'm just writing the entire spreadsheet into the document row. i know that previously, we tried to parse the document here, and save it with -- column headings? not sure what the previously format is. i'm wondering if this makes sense. i do think we can transform the sheet on read when producing treasury reports, and keeping it the rows/columns just as input would make validation a little easier, we can point the user to the exact cells that fail to validate by the row/column number

Reply:
Oh, interesting!

I think at some point we're going to have to do the work we were doing here, regardless.  In the old system, the "documents" table was actually a collection of "rows".  Ultimately, we're going to need to parse the spreadsheets and extract rows so that we can aggregate across uploads at write time.

I think my first instinct is to keep doing that and keep this table.  Does that cause problems for your new upload approach?

In some places in the code, these are called "records" instead of "documents" — that might be a better name for the table, if we keep it.